File: VKHCG/02-Krennwallner/01-Retrieve/Retrieve_DE_Billboard_Locations.r
###########################################
rm(list=ls()) #will remove ALL objects
###########################################
if (Sys.info()['sysname'] == "Windows") 
  {
    BaseSet = "C:/VKHCG"
    setwd(BaseSet)
  } else {
    BaseSet = paste0("/home/" , Sys.info()['user'] , "/VKHCG")
    setwd(BaseSet)
  }
###########################################
Base=getwd()
FileDir= paste0(Base,'/02-Krennwallner/01-Retrieve/01-EDS/01-R')
###########################################
library(readr)
library(data.table)
###########################################
FileName=paste0(Base,'/02-Krennwallner/00-RawData/DE_Billboard_Locations.csv')
###########################################
DE_Billboard_Locations <- read_csv(FileName,
                                   col_types = cols(
                                     Country = col_character(), 
                                     ID = col_integer(), 
                                     Latitude = col_double(), 
                                     Longitude = col_double(), 
                                     PlaceName = col_character()
                                   ), na = "empty")
###########################################
setnames(DE_Billboard_Locations,'PlaceName','Place.Name')
###########################################
setorder(DE_Billboard_Locations,Latitude,Longitude)
###########################################
FileNameOut=paste0(FileDir,'/Retrieve_DE_Billboard_Locations.csv')
fwrite(DE_Billboard_Locations, FileNameOut)
###########################################
View(DE_Billboard_Locations)
###########################################

File: VKHCG/02-Krennwallner/01-Retrieve/Retrieve-DE-Billboard-Locations.py
################################################################
# -*- coding: utf-8 -*-
################################################################
import sys
import os
import pandas as pd
################################################################
InputFileName='DE_Billboard_Locations.csv'
OutputFileName='Retrieve_DE_Billboard_Locations.csv'
Company='02-Krennwallner'
################################################################
if sys.platform == 'linux': 
    Base=os.path.expanduser('~') + '/VKHCG'
else:
    Base='C:/VKHCG'
print('################################')
print('Working Base :',Base, ' using ', sys.platform)
print('################################')
################################################################
Base='C:/VKHCG'
sFileName=Base + '/' + Company + '/00-RawData/' + InputFileName
print('Loading :',sFileName)
IP_DATA_ALL=pd.read_csv(sFileName,header=0,low_memory=False,
  usecols=['Country','PlaceName','Latitude','Longitude'])

IP_DATA_ALL.rename(columns={'PlaceName': 'Place_Name'}, inplace=True)
################################################################
sFileDir=Base + '/' + Company + '/01-Retrieve/01-EDS/02-Python'
if not os.path.exists(sFileDir):
    os.makedirs(sFileDir)

ROUTERLOC = IP_DATA_ALL.drop_duplicates(subset=None, keep='first', inplace=False)

print('Rows :',ROUTERLOC.shape[0])
print('Columns :',ROUTERLOC.shape[1])

sFileName2=sFileDir + '/' + OutputFileName
ROUTERLOC.to_csv(sFileName2, index = False)

################################################################
print('### Done!! ############################################')
################################################################